From c787366b80d240e79253759882b0eff315997730 Mon Sep 17 00:00:00 2001
From: Suneel Garapati <sgarapati@caviumnetworks.com>
Date: Tue, 14 Aug 2018 15:25:56 -0700
Subject: [PATCH 0816/1239] net: octeontx2: cgx_intf: skip error check for
 shutdown command

For CGX_CMD_INTF_SHUTDOWN, skip check on errors and clear
the scratch registers.

Signed-off-by: Nithya <nithya.mani@caviumnetworks.com>
Signed-off-by: Suneel Garapati <sgarapati@caviumnetworks.com>
---
 drivers/net/cavium/octeontx2/cgx_intf.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/net/cavium/octeontx2/cgx_intf.c b/drivers/net/cavium/octeontx2/cgx_intf.c
index 13bcb3d66f..6dcf6a90e2 100644
--- a/drivers/net/cavium/octeontx2/cgx_intf.c
+++ b/drivers/net/cavium/octeontx2/cgx_intf.c
@@ -123,6 +123,9 @@ int cgx_intf_req(u8 cgx, u8 lmac, u8 cmd, u64 *rsp)
 
 	set_ownership(cgx, lmac, CGX_OWN_NON_SECURE_FW);
 
+	if (cmd == CGX_CMD_INTF_SHUTDOWN)
+		goto error;
+
 	if (scr0.s.evt_sts.evt_type != CGX_EVT_CMD_RESP) {
 		printf("%s received async event instead of cmd resp event\n",
 			__func__);
@@ -141,17 +144,14 @@ int cgx_intf_req(u8 cgx, u8 lmac, u8 cmd, u64 *rsp)
 		err = -1;
 		goto error;
 	}
-	
+
+error:
 	/* clear ownership and ack */
 	scr0.s.evt_sts.ack = 0;
 	cgx_wr_scr0(cgx, lmac, scr0.u);
 	set_ownership(cgx, lmac, CGX_OWN_NONE);
 
-	*rsp = scr0.u;
-
-error:
-	if (err)
-		*rsp = 0;
+	*rsp = err ? 0 : scr0.u;
 
 	return err;
 }
-- 
2.29.0

